plugin/trino-lancedb/src/test/java/io/trino/plugin/lance/TestLanceMetadata.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.lance;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.Resources;
+import io.trino.plugin.lance.internal.LanceReader;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.TableNotFoundException;
+import java.net.URL;
+import java.util.Optional;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import static io.trino.testing.TestingConnectorSession.SESSION;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_METHOD;
+
+
+@TestInstance(PER_METHOD)
+public class TestLanceMetadata
+{
+    private static final LanceTableHandle TEST_TABLE_1_HANDLE = new LanceTableHandle("example", "test_table1");
+    private static final LanceTableHandle TEST_TABLE_2_HANDLE = new LanceTableHandle("example", "test_table2");
```

Comment:
currently i only have 1 test table copied from lance. would be nice if we have more with various data schema. please advice if there's any existing test resource i can copy over

Reply:
https://github.com/lancedb/lance/tree/main/test_data We have test dataset.